### DISCUSSION TITLE
🐞 FastAPI doesn't support list of literals in query params

### DISCUSSION THREAD

**Discussion by jd-solanki** [fastapi/fastapi#12380] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Literal

from fastapi import FastAPI

app = FastAPI()

type Status = Literal["new", "in_progress", "completed", "cancelled"]


@app.get("/")
async def single_status(status: Status):
    return {"Hello": status}


@app.get("/list")
async def multiple_status(status: list[Status]):
    return {"Hello": status}
```

### Description

- Run above app
- Check `/` has proper UI that allows selecting from one of the literals
- Check `/list` which breaks the OpenAPI/FastAPI

**Secreenshot:**
![image](https://github.com/user-attachments/assets/f0fbf2ce-e382-4356-b0a6-a83bd463bc7d)

It should allow inserting multiple status

### Operating System

Linux

### Operating System Details

Ubuntu 20.04.6 LTS

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

Python 3.12.5

### Additional Context

It should be something like this: 
![image](https://github.com/user-attachments/assets/b8dd0ee4-4f9c-44d5-b67c-a5d635c2267b)

**Comment by JavierSanchezCastro** [comment, marked as answer]
Add Query:

```python3
from typing import Literal, Annotated

from fastapi import FastAPI, Query

app = FastAPI()

type Status = Literal["new", "in_progress", "completed", "cancelled"]


@app.get("/")
async def single_status(status: Status):
    return {"Hello": status}


@app.get("/list")
async def multiple_status(status: Annotated[list[Status], Query()]):
    return {"Hello": status}
```
